src/ray/ray_config.h

```diff
@@ -35,9 +36,13 @@ class RayConfig {
 /// -----------Include ray_config_def.h to set config items.-------------------
 /// A helper macro that helps to set a value to a config item.
 #define RAY_CONFIG(type, name, default_value) \
-  if (pair.first == #name) { name##_ = pair.second; continue; }
+  if (pair.first == #name) {                  \
+    std::istringstream stream(pair.second);   \
+    stream >> name##_;                        \
+    continue;                                 \
+  }
```

Comment:
Another approach to reserve the `;` for every `RAY_CONFIG ()` statement is to add a `RAY_IGNORE()` at the end of this macro like:
```C++
#define RAY_CONFIG(type, name, default_value) \
 if (pair.first == #name) {                  \
    std::istringstream stream(pair.second);   \
    stream >> name##_;                        \
    continue;                                 \
  }                                           \
  RAY_IGNORE(nothing)
```

But I don't know if this way is good enough.

Reply:
That way looks a bit wired. We can wait for other developers' comments.

Reply:
I don't think this is wired since it's similar to `RAY_LOG()`.

Reply:
I have a slight preference for requiring a `;` after the macro also. If you add `RAY_IGNORE` at the end, then I'd suggest including a comment to explain its purpose.

Reply:
OK, will do.

Reply:
Adding `RAY_IGNORE` looks fine to me as well.

Reply:
`RAY_IGNORE_EXPR(0)` can only be used in functions. It is not allowed to use `RAY_IGNORE_EXPR(0)` when a class is defined. I used another way to may the `;` required.